syntax = "proto3";

package bavs;

import "validate/validate.proto";

message BAVSParameter {
    string name = 1;
    string value = 2;
    string type = 3;
    string default_value = 4;
    bool has_default_value = 5;
}

message Upstream {
    string full_hostname = 1;
    int32 port = 2;
    string path = 3;
    string method = 4;
    int32 total_attempts = 5;
    string wf_tid = 6;
}

message BAVSFilter {
    Upstream inbound_upstream = 1;
    repeated Upstream forward_upstreams = 2;
    Upstream flowd_upstream = 3;
    Upstream shadow_upstream = 4;
    repeated Upstream error_gateway_upstreams = 5;
    string wf_did = 6;
    repeated string headers_to_forward = 7;
    bool closure_transport = 8;
    repeated BAVSParameter input_params = 9;
    repeated BAVSParameter output_params = 10;
    string wf_did_header = 11;
    string wf_tid_header = 12;
    string wf_iid_header = 13;
}
